docs/strings.md

```diff
@@ -0,0 +1,529 @@
+ * Support all string operations.
+   Some of these, such as locale-aware collation, are incredibly complex.
+ * Interoperation with other systems that place constraints on string representation (e.g. system call arguments).
+
+Many existing languages have suffered from the definition of a character changing over the lifetime of the language.
+When C was introduced the consensus that a character was 8 bits was still fairly new, with older machines often supporting 6-bit characters.
+The developers of C used Latin-alphabet languages and multi-byte encodings were later retrofitted to C in somewhat awkward ways, breaking the abstraction that one `char` is one `character`.
+
+The OpenStep specification and Java came at a time when the Unicode 1.0 specification was current and there appeared to be consensus that 16 bits was sufficient to represent any character in any language.
+This consensus changed in 1996 with the release of the Unicode 2.0 specification, expanding the encoding space to up to 1,112,064 code points.
+The current version of the Unicode specification, 13.0, uses just over 10% of this encoding space and even with the amount wasted on emoji it seems unlikely that the total space will be exhausted and so Verona is likely to be in a in a fortunate position of being able to define a string API without the definition of a character changing substantially over the lifetime of the language.
+
+It is incredibly unlikely; however, that a Verona string implementation will contain native Verona code that implements the full Unicode specification in an early release.
+The specification of the [Unicode collation algorithm](https://www.unicode.org/reports/tr10/) alone runs to several hundred pages of text.
+A string design will therefore need to be efficient when interoperating with a sandboxed C++ library, such as ICU.
+
+Who cares about strings?
+------------------------
+
+Verona aims to be an *infrastructure programming language*.
+This category encompasses target use cases such as device drivers, distributed databases, online translation engines, and so on.
+Some of these will not have any user-facing component and so will use strings as identifiers (for example, dictionary keys, path components, error indicators) without the need for any localisation.
+Others will use strings to present text to a human and will need rich unicode processing.
+Our goal should be to support both use cases with the same set of abstractions but avoid projects in the former category needing to pay any performance or code-size penalty relative to a simple array-of-probably-ASCII-bytes string representation.
+
+A lot of complex Verona systems are likely to include components that are at both extremes in terms of their use of strings.
+We should not require a global decision about how strings are represented, nor should we penalise modules that require only simple strings.
```

Comment:
I think it's not quite that.  Core standard library things should all take a `String` interface.  We will provide an `AsciiString` concrete class and one or more unicode packages that provide richer strings.  If you don't care about anything other than ASCII, you're free to avoid using the other packages and just use ASCII everywhere.